public/utils/constants.ts

```diff
@@ -317,3 +317,5 @@ const LocalCluster: DataSourceOption = {
 };
 
 export const dataSourceObservable = new BehaviorSubject<DataSourceOption>(LocalCluster);
+
+export const DATA_SOURCE_NOT_SET_ERROR = 'Data source is not set';
```

Comment:
Not blocking; but I believe there's a best practice to end error text with a period (`.`).

Reply:
We don't show this to the user, just check for this error